src/Juvix/Core/EAC/Check.hs

```diff
@@ -1,12 +1,13 @@
-module Juvix.EAC.Check where
+module Juvix.Core.EAC.Check where
 
 import qualified Data.Map.Strict as Map
-import Juvix.EAC.EAC
-import Juvix.EAC.Solve
-import Juvix.EAC.Types
+import Juvix.Core.EAC.ConstraintGen
+import Juvix.Core.EAC.Solve
+import Juvix.Core.EAC.Types
+import Juvix.Core.Erased.Types
 import Juvix.Library hiding (link, reduce)
 
-validEal ∷ Term → TypeAssignment → IO (Either Errors (RPT, ParamTypeAssignment))
+validEal ∷ ∀ primTy primVal. (Eq primTy) ⇒ Term primVal → TypeAssignment primTy → IO (Either (Errors primTy primVal) (RPT primVal, ParamTypeAssignment primTy))
```

Comment:
Should probably make this multiple lines, it gets past 159 characters long

Reply:
Yup, updated.